### PR TITLE
#3: Updated user guide

### LINKED ISSUE #3: Add user guide

### PR DESCRIPTION
**Pull request by morazow**
Fixes #3

### CODE REVIEW

File: doc/user_guide/user_guide.md [hunk 1]
```diff
@@ -1,5 +1,111 @@
...
+- [Job Parameters](#job-parameters)
+- [Data Mapping](#data-mapping)
+
+## Preparing Required Services
+
+To use the AWS Glue Studio connector, you should prepare the following steps which then will be used setting up the connector.
+
+### Creating IAM Role
+
+We are going to create an IAM Policy Role with required permissions that will act on our behalf when using AWS Glue Studio Connector. Go to ["IAM Service"](https://console.aws.amazon.com/iam/) &rarr; from navigation page choose "Roles" &rarr; click "Create Role" on the top right corner. Please follow these steps:
```
Comment (kaklakariada) [inline doc/user_guide/user_guide.md:16]: As an alternative we could also show instructions using the CLI. Then we could specify the roles & permissions directly.

File: doc/user_guide/user_guide.md [hunk 2]
```diff
@@ -1,5 +1,111 @@
...
+- Type "S3" to Filter Policies and press enter, and then select `AmazonS3FullAccess` policy
+- Type "Glue" and press enter, and then select `AWSGlueServiceRole` policy
+- Similarly, search for and add `AmazonEC2ContainerRegistryReadOnly` policy
+- These are the default policy roles, however, please change them accordingly
+- Then click next, and provide a descriptive Role name and Role description
+- Add tags, review and click "Create role" button
+
+![Exasol AWS Glue Studio IAM Role](img/iam_role.png)
+
+Please read about the AWS Glue Studio [documentation on minimum required permissions](https://docs.aws.amazon.com/glue/latest/ug/setting-up.html#getting-started-min-privs-connectors).
```
Comment (kaklakariada) [inline doc/user_guide/user_guide.md:30]:
```suggestion
```

File: doc/user_guide/user_guide.md [hunk 3]
```diff
@@ -1,5 +1,111 @@
...
+
+## Setting up the Connector
+
+To setup the connector you will first need to login to your AWS account. Then, go to the Exasol AWS Glue Studio [connector marketplace](https://aws.amazon.com/marketplace/pp/prodview-2mlebkenfsg7o) page, and click on "Continue to Subscribe" button.
+
+Once you accepted the terms and conditions, click on "Usage" tab, go to "View usage instructions" and click to "Activate the Glue connector from AWS Glue Studio" link.
```
Comment (kaklakariada) [inline doc/user_guide/user_guide.md:36]:
```suggestion
Once you accepted the terms and conditions, click on "Usage" tab, go the "View usage instructions" and click to "Activate the Glue connector from AWS Glue Studio" link.
```

File: doc/user_guide/user_guide.md [hunk 4]
```diff
@@ -1,5 +1,111 @@
...
+
+It will bring you into AWS Glue Studio service. Please add a unique connection name.
+
+![Exasol AWS Glue Studio Connection Name](img/glue_connection.png)
+
+Then click "Create Connection and Activate Connector" button.
+
+#### Confirm That Connector and Connection are Created
+
+To confirm that the connector is activated and the corresponding connection is created, go to "AWS Glue Studio" &rarr; "Connectors". You should see the connector and connection are created as in following images.
```
Comment (kaklakariada) [inline doc/user_guide/user_guide.md:50]:
```suggestion
To confirm that the connector is activated and the corresponding connection is created, go to "AWS Glue Studio" &rarr; "Connectors". You should see that the connector and connection are created as shown in following images:
```

File: doc/user_guide/user_guide.md [hunk 5]
```diff
@@ -1,5 +1,111 @@
...
+
+![Exasol AWS Glue Studio Connection](img/connection.png)
+
+### Creating a Job
+
+Let's now create a AWS Glue Studio Job that accesses Exasol database. Go to "AWS Glue Studio" &rarr; "Jobs", select "Visual with a blank canvas" and click on the "Create" button.
+
+![Exasol AWS Glue Studio Job Create](img/job_create.png)
+
+This creates a black canvas. From the "Source" dropdown menu select the "Exasol Connector for AWS Glue" and from the "Target" drowdown menu select "Amazon S3". This way, we are creating a simple job that will read from Exasol database and store the results into an S3 location.
```
Comment (kaklakariada) [inline doc/user_guide/user_guide.md:62]:
```suggestion
This creates a blank canvas. From the "Source" dropdown menu select the "Exasol Connector for AWS Glue" and from the "Target" drowdown menu select "Amazon S3". This way, we are creating a simple job that will read from Exasol database and store the results into an S3 location.
```

File: doc/user_guide/user_guide.md [hunk 6]
```diff
@@ -1,5 +1,111 @@
...
+
+![Exasol AWS Glue Studio Canvas](img/canvas.png)
+
+#### Update Source Parameters
+
+We can now setup our source node parameters. Click on the "Data Source", for the connection select the "Connection Name" we created earlier. Then, click on "Connection options" &rarr; "Add new option", and add required key value parameters. You can read about them on [Job Parameters](#job-parameters) section.
```
Comment (kaklakariada) [inline doc/user_guide/user_guide.md:68]:
```suggestion
We can now setup our source node parameters. Click on the "Data Source", for the connection select the "Connection Name" we created earlier. Then, click on "Connection options" &rarr; "Add new option", and add required key value parameters. You can read about them in the [Job Parameters](#job-parameters) section.
```

File: doc/user_guide/user_guide.md [hunk 7]
```diff
@@ -1,5 +1,111 @@
...
+| `username`            |                    |    ✓     | `sys`                                           |
+| `password`            |                    |    ✓     | `3wcyPtZuYX7p4gTeyJyv`                          |
+| `awsAccessKeyId`      |                    |    ✓     | `AKIA5NZSQR...`                                 |
+| `awsSecretAccessKey`  |                    |    ✓     | `cwkpgmTwtEZh0SC...`                            |
+| `s3Bucket`            |                    |    ✓     | `exasol-retail-frankfurt`                       |
+| `table`               |                    |          | `AWS_GLUE.RETAIL`                               |
+| `query`               |                    |          | `SELECT product_id, price FROM AWS_GLUE.RETAIL` |
+| `numPartitions`       | `8`                |          | `16`                                            |
+| `awsRegion`           | `us-east-1`        |          | `eu-central-1`                                  |
+| `awsEndpointOverride` | `default endpoint` |          | `s3.custom-endpoint.com`                        |
```
Comment (kaklakariada) [inline doc/user_guide/user_guide.md:104]:
```suggestion
| `awsEndpointOverride` | (default endpoint) |          | `s3.custom-endpoint.com`                        |
```

File: doc/user_guide/user_guide.md [hunk 8]
```diff
@@ -1,5 +1,111 @@
...
+| `s3PathStyleAccess`   | `false`            |          | `true`                                          |
+| `useSsl`              | `true`             |          | `false`                                         |
+
+Either one of `table` or `query` parameters should be provided when reading data from Exasol database.
```
Comment (kaklakariada) [inline doc/user_guide/user_guide.md:108]:
```suggestion
Either one of `table` or `query` parameters should be provided when reading data from an Exasol database.
```

File: doc/user_guide/user_guide.md [hunk 9]
```diff
@@ -1,5 +1,111 @@
...
+
+We are going to create an IAM Policy Role with required permissions that will act on our behalf when using AWS Glue Studio Connector. Go to ["IAM Service"](https://console.aws.amazon.com/iam/) &rarr; from navigation page choose "Roles" &rarr; click "Create Role" on the top right corner. Please follow these steps:
+
+- Ensure that "AWS service" is selected
+- For "Use cases for other AWS services" search for "Glue" and select "Glue"
+- Click "Next" to add permissions
+- Type "S3" to Filter Policies and press enter, and then select `AmazonS3FullAccess` policy
+- Type "Glue" and press enter, and then select `AWSGlueServiceRole` policy
+- Similarly, search for and add `AmazonEC2ContainerRegistryReadOnly` policy
+- These are the default policy roles, however, please change them accordingly
```
Comment (kaklakariada) [inline doc/user_guide/user_guide.md:24]:
```suggestion
- **Note:** These are the default policy roles, however, please change them accordingly. See the AWS Glue Studio [documentation on minimum required permissions](https://docs.aws.amazon.com/glue/latest/ug/setting-up.html#getting-started-min-privs-connectors).
```

**Review by kaklakariada — CHANGES_REQUESTED** [review]

File: doc/user_guide/user_guide.md (hunk 1 above)
Comment (morazow) [reply doc/user_guide/user_guide.md:16]: Sure, it would be nice. I created separate issue #48 for it.

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**